bloomberg/ipydatagrid
# 🐛 Resolved issue for data getter when datagrid is empty

*Issue number of the reported bug or feature request: #322*

**Describe your changes**
Added an if statement to check the __data_ traitlet and see if its "data" value is empty within the data getter method. If it is empty, then construct an empty dataframe from the schema fields so final_df will be constructed from that df.

**Testing performed**
Re-ran the code in the issue.

1. Create test DataGrid.

```python
df = pd.DataFrame([1], columns=["a"])
grid = DataGrid(df)
grid
```

We get this output:
![image](https://user-images.githubusercontent.com/79875276/169362474-9771cb39-f60b-46a3-8ca5-d03c17d247b6.png)

2. Use data getter:

```python
grid.data
```

![image](https://user-images.githubusercontent.com/79875276/169363472-35fa7177-f8cf-46f2-8c85-a79af6a120c2.png)

3. Now drop all the rows in the dataframe and set using the data setter.

```python
grid.data = grid.data.drop(0)
```

We see that the DataGrid UI object reflects those changes (we only have the columns):

![image](https://user-images.githubusercontent.com/79875276/169362691-8dfc317a-031d-467a-8930-248fbbe32bef.png)

4. Now call the data getter again:

```python
grid.data
```

5. Data getter now returns appropriate dataframe.

![image](https://user-images.githubusercontent.com/79875276/196738938-6b2fa8e5-52ae-486a-b0e1-4dd02fa83d4d.png)


**Additional context**
No additional information.
